tuner/tuner/libtuner.py

```diff
@@ -221,6 +221,53 @@ def validate_devices(user_devices: list[str]) -> None:
         )
 
 
+def validate_benchmark_results(
+    benchmark_results: list[BenchmarkResult],
+) -> list[BenchmarkResult]:
```

Comment:
Based on the function name alone, it's impossible to tell what this code does... Could we call it something like: `get_valid_benchmark_results` and add a one-line docstring explaining what is considered valid?